tracing/tracers/basic/basic.go

```diff
@@ -4,13 +4,25 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	basic "github.com/opentracing/basictracer-go"
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
-func InitTracer(opts []string) (opentracing.Tracer, error) {
+type CloseableTracer interface {
+	opentracing.Tracer
+	Close()
+}
+
+type BasicTracer struct {
+	tracer opentracing.Tracer
```

Comment:
I think here you can embed `opentacing.Tracer` to avoid implementing method overrides
```go
type BasicTracer struct {
	opentracing.Tracer
	...
}
```

Reply:
I don't really like embedding, but it seems to work in this case.